### ISSUE TITLE
Custom modules controller action not working in rc2.

### ISSUE THREAD

**Issue by cedcoss** [magento/magento2#2381]
Custom modules controller actions were working in the magento2 rc version. But after magento2 rc2 these action are giving 404 error.

**Comment by ivanweiler** [comment]
Same here. I think it's the autoloader problem, \_empty\_ declaration in composer.json is causing this:
https://github.com/magento/magento2/blob/2335247d4ae2dc1e0728ee73022b0a244ccd7f4c/composer.json#L226

In RC2 `/vendor/composer/autoload_namespaces.php` looks like:
```
return array(
    '_empty_' => array($baseDir . '/app/code'),
..
```
If you change '_empty_' to '' (as it was in RC1), custom modules work, since autoloader can find them.

Controller doesn't work because is_subclass_of() in \Magento\Framework\App\Router\ActionList always returns false since it can't find/autoload controller in custom modules!!

To summarize, custom modules don't work in M2 RC2 because of this, most likely every module that's not in Magento namespace !!

**Comment by vrann** [comment]
Hi,
thanks for reporting the issue. 

It was an issue in publication script which was used to package RC2. As @ivanweiler mentioned, it created  '_empty_' node in the composer.json which registers PSR for custom modules.

We discovered it during regression testing and fixed. You should get correct composer.json in nearest release.

**Comment by pradeeprcs** [comment]
@ivanweiler :+1: thanks but can you tell me what is in RC1 because i dont have RC1 code so please share "_empty_" in RC1which will be help full or else can you mention what step wise with code to make custom module  working 
and also when i run update cli it not install my custom tables that is its not entering to Setup/install file but it make entry in setup_module and config.php file

**Comment by davidalger** [comment]
Duplicate of #2379 

The "solution" to this so you can get it working in RC2 is to change this…

        "psr-0": {
            "_empty_": "app/code/"
        },

to this…

        "psr-0": {
            "": "app/code/"
        },

in the composer.json file and then run `composer install` to re-generate the auto-load files

**Comment by davidalger** [comment]
This is fixed in the 2.0 GA.
